### ISSUE TITLE
Merging more than 64 files fails

### ISSUE THREAD

**Issue by CreateRandom** [DavidFirth/pdfjam#14]
Hello,

I'm trying to merge around 300 single page PDFs into a single file. I used the following command

`pdfjam *.pdf --nup '2x1' --landscape --outfile merged.pdf --no-tidy`

This fails with a `The call to /home/username/bin/pdflatex resulted in an error.`

The last line in the log file reads: `!pdfTeX error: pdflatex (file /var/tmp/pdfjam-bg2gZJ/source-64.pdf): reading im
age file failed
 ==> Fatal error occurred, no output PDF file produced!`

Inspecting the temporary PDF file shows it is empty, as are all the temporary files whose index is greater than 64. I'm using the most recently released version of pdfjam on Ubuntu 19.10 (copied the binary directly from here). I could also observe this behaviour with version 2.08 that is shipped in the texlive-utils.

Is this some known limitation or possibly just some variable that overflows? Thanks for looking into this.

**Comment by DavidFirth** [comment]
I have not seen this myself --- it's not a known limitation (to me, anyway).  It looks like it could _maybe_ be a limitation of `pdflatex` (on your system?).  Two things you might try, if using v3.02 or later of `pdfjam`:

- specify a different engine fro `pdflatex`, by for example `--latex /usr/bin/xelatex` or `--latex /usr/bin/lualatex`; or
- see the FAQ (in the README) about 'Too many open files'.  (A bit of a long shot, perhaps)

I hope that helps.  Please do let me know if either of those suggestions works for you.

**Comment by CreateRandom** [comment]
Thanks for your quick and helpful reply! I tried switching out the backend and inserting the preamble, but to no avail. Only then I noticed the reasons the files were empty: I had assumed that something had gone wrong during copying into the temp folder, but these are not copies in there, but links. I had a couple files that were empty to due an unrelated error during PDF creation. Due to the magic number 64, it just seemed more likely that the error was on the end of pdfjam, but it really wasn't. Great tool by the way, really appreciate it!
